src/modules/messagePreview/module.js

```diff
@@ -25,27 +30,43 @@ subscriptions.set('messageCreate', async (client, message) => {
         if (
             !msg.channel
                 .permissionsFor(message.author)
-                ?.has([Permissions.FLAGS.VIEW_CHANNEL, Permissions.FLAGS.READ_MESSAGE_HISTORY])
+                ?.has([
+                    PermissionsBitField.Flags.ViewChannel,
+                    PermissionsBitField.Flags.ReadMessageHistory,
+                ])
         )
             return;
         if (!msg.member) await msg.guild.members.fetch(msg.author);
 
-        const embed = new MessageEmbed()
-            .setAuthor(msg.member.displayName, msg.member.displayAvatarURL({ dynamic: true }))
-            .setDescription(msg.content)
+        const embed = new EmbedBuilder()
+            .setAuthor({
+                name: msg.member.displayName,
+                iconURL: msg.member.displayAvatarURL(),
+                url: msg.url,
+            })
             .setColor(msg.member.displayColor)
-            .setFooter(`Requested by ${message.member.displayName}`)
-            .setTimestamp(msg.createdTimestamp);
+            .setFooter({ text: `Requested by ${message.member.displayName}` })
+            .setTimestamp(msg.createdTimestamp)
+            .setURL(msg.url);
 
+        const additionalEmbeds = [];
+        if (msg.content) embed.setDescription(msg.content);
         if (msg.attachments.size) {
-            embed.setImage(msg.attachments.first().url);
-            embed.addField(
-                'Attachments:',
-                msg.attachments.map((a) => `[${a.name}](${a.url})`).join('\n')
-            );
+            const imgs = msg.attachments.filter((a) => a.contentType.startsWith('image/'));
+            Array.from(imgs.entries()).forEach(([, a], i) => {
+                console.log(a, i);
+                if (i === 0) return embed.setImage(a.url);
+                if (i > 3) return null;
+                return additionalEmbeds.push(new EmbedBuilder().setImage(a.url).setURL(msg.url));
+            });
+            embed.addFields({
+                name: 'Attachments:',
+                value: msg.attachments.map((a) => `[${a.name}](${a.url})`).join('\n'),
+            });
```

Comment:
what shall we do (with the drunken sailor) with non-image files e.g. text or audio?

Reply:
we filter for image/ kind of attachments above. but ignore any attachments which are not images, like text files or audios. this results in more empty embeds

Reply:
No this uses msg.attachments and not imgs so this works as intended.